packages/xo-server/src/xo-mixins/backups-ng/index.js

```diff
@@ -150,6 +150,43 @@ export default class BackupNg {
         // different than the VMs in the job itself.
         let vmIds = data?.vms ?? extractIdsFromSimplePattern(vmsPattern)
         if (vmIds === undefined) {
+          const poolPattern = vmsPattern.$pool
+
+          // Log a failure task when a pool contained in the smart backup
+          // pattern doesn't exist
+          if (poolPattern !== undefined) {
+            let poolIds = extractIdsFromSimplePattern({ id: poolPattern })
+            if (poolIds === undefined && Array.isArray(poolPattern.__and)) {
+              poolIds = []
+              poolPattern.__and.forEach(pattern => {
+                const ids = extractIdsFromSimplePattern({ id: pattern })
+                if (ids !== undefined) {
+                  poolIds = [...poolIds, ids]
+                }
+              })
```

Comment:
As I said, I do believe this is incorrect.

What does contain `$pool.__and`?

Reply:
An array.

Reply:
What does the array contain?

Reply:
Pool patterns.

Reply:
Why do these pool patterns look like?

Didn't you forget to take a `__not` into account?

Reply:
We don't need to check this case because if the pool doesn't exist, its VMs will not exist.

Reply:
Understood.